packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +829,385 @@ def to_json(self):
+                    indent=indent + 1,
+                    not_first=True,
+                    name=name,
+                    inner_model=inner_model,
+                    no_list=no_list,
+                )
+                inner_model = format_punctuation(self, json_request, indent, inner_model, i)
+    return not_first,indent,inner_model
+
+def format_punctuation(self, json_request, indent, inner_model, i):
+    if i == "str":
+        pass
+    elif inner_model and isinstance(json_request[i], list):
+        if isinstance(json_request[i], list) and len(json_request[i])>1:
+                inner_model.append(Token(" ", TokenKind.Newline))
+                inner_model.append(
+                                Token(" " * (indent * 4), TokenKind.Whitespace)
+                            )
+                inner_model.append(Token("}[];", TokenKind.Comment))
+        elif isinstance(json_request[i],list) and isinstance(json_request[i][0],dict):
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("}[];", TokenKind.Comment))
+        elif len(json_request[i])>1:
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+    elif isinstance(json_request[i], list) and indent > 4:
+        if isinstance(json_request[i][0], dict) :
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "}[];", None)
+        if len(json_request[i])==1:
+            pass
+        else:
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "}[];", None)
+    else:
+        if indent == 5 and inner_model:
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+            if self.inner_model != inner_model:
+                self.inner_model +=inner_model
+            else: self.inner_model = inner_model
+            inner_model = []
+        elif inner_model:
+            inner_model.append(Token(" ", TokenKind.Newline))
+            inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+        else:
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "};", None)
+    return inner_model
+
+def list_format(self, json_request, yaml, not_first, indent, name, inner_model):
```

Comment:
The value for the `not_first` param doesn't get used in the function body -- not sure if intentional.